administrator/components/com_media/resources/scripts/components/browser/items/item.js

```diff
@@ -42,9 +42,29 @@ export default {
          * @param event
          */
         function handleClick(event) {
-            var e = createEvent('onMediaFileSelected');
-            e.item = item;
-            window.parent.document.dispatchEvent(e);
+            const rootPath = Joomla.getOptions('com_media').fileBaseRelativeUrl;
+            const cloudRootPath = Joomla.getOptions('com_media').fileBaseRelativeUrl; //@todo return the cloud root...
+            const isCloud = false; //@todo return true if file is on the cloud
```

Comment:
this is not need, we treat every adapter in the same way, either you used local or whatever path is resolved by the API call

Reply:
where is this info propagated in the front end?

Reply:
it is in the options, but you don't need to separate clouds and non clouds
we treat all in the same way. Also user can have multiple folders in Local Adapter

Reply:
@kasvith don't mix PHP with JS, what you are talking about is PHP, what is needed here in the JS scope is a var with the root path (the first part that will form the relative or absolute path). Is such var stored somewhere? If not it needs to be. Best way is to attach this info to the property of the item

Reply:
no. it is not. 
actually there is no fileBaseRelativeUrl anymore. I will remove it as we can have multiple folders in MM